client/src/components/SummaryCalculation.ts

```diff
@@ -0,0 +1,83 @@
+import _ from "lodash";
+import helpers from "util/helpers";
+import { AddressLocation, AddressRecord, SummaryStatsRecord } from "./APIDataTypes";
+
+export const getTopFiveContactsInPortfolio = (addrs: AddressRecord[]) => {
+  const allContactNames = addrs
+    .map((a) => a.ownernames?.map((ownername) => ownername.value) || [])
+    .flat();
+
+  return helpers.getMostCommonElementsInArray(allContactNames, 5);
+};
+
+export const extractLocationDataFromAddr = (addr: AddressRecord): AddressLocation => {
+  return (({ boro, housenumber, lat, lng, streetname }) => ({
+    boro,
+    housenumber,
+    lat,
+    lng,
+    streetname,
+  }))({
+    ...addr,
+  });
+};
+
+export const calculateAggDataFromAddressList = (addrs: AddressRecord[]): SummaryStatsRecord => {
+  const bldgs = addrs.length;
+  const units = _.sumBy(addrs, (a) => a.unitsres || 0);
+
+  const totalopenviolations = _.sumBy(addrs, (a) => a.openviolations);
+  const totalviolations = _.sumBy(addrs, (a) => a.totalviolations);
+
+  const totalevictions = _.sumBy(addrs, (a) => a.evictions || 0);
+
+  const totalrsgain = _.sumBy(addrs, (a) => (a.rsdiff && a.rsdiff > 0 ? a.rsdiff : 0));
```

Comment:
do you need the `&& a.rsdiff > 0`? i.e. are there any cases where this number is negative? The cases are:
- rs is null (in which case you want to reset to 0)
- rs is 0 (in which case who cares if it gets summed, since it's just 0)
- rs > 0 (in which you want to sum it)
- rs < 0 (in which case I can imagine you wanting to explicitly reset it to 0, but does that happen?)

Reply:
hey @samaratrilling, so there are many cases where rsdiff is negative, specifically when there is a drop in rs units from 2007 to 2019 for a given building. The idea behind my function is that we want to filter out the negative values so we can see just a tally of how many units were gained across the portfolio. I can add a comment here to make it more clear-does that make sense?